pkg/controller/acicontainersoperator.go

```diff
@@ -97,35 +99,116 @@ func NewAciContainersOperator(
 	)
 
+	aci_deployment_informer := cache.NewSharedIndexInformer(
+		&cache.ListWatch{
+			ListFunc: func(options metav1.ListOptions) (runtime.Object, error) {
+				return k8sclient.AppsV1().Deployments(os.Getenv("SYSTEM_NAMESPACE")).List(options)
+			},
+			WatchFunc: func(options metav1.ListOptions) (watch.Interface, error) {
+				return k8sclient.AppsV1().Deployments(os.Getenv("SYSTEM_NAMESPACE")).Watch(options)
+			},
+		},
+		&appsv1.Deployment{},
+		0,
+		cache.Indexers{},
+	)
+
+	aci_daemonset_informer := cache.NewSharedIndexInformer(
+		&cache.ListWatch{
+			ListFunc: func(options metav1.ListOptions) (runtime.Object, error) {
+				return k8sclient.AppsV1().DaemonSets(os.Getenv("SYSTEM_NAMESPACE")).List(options)
+			},
+			WatchFunc: func(options metav1.ListOptions) (watch.Interface, error) {
+				return k8sclient.AppsV1().DaemonSets(os.Getenv("SYSTEM_NAMESPACE")).Watch(options)
+			},
+		},
+		&appsv1.DaemonSet{},
+		0,
+		cache.Indexers{},
+	)
+
 	controller := &Controller{
 		Logger:    log.NewEntry(log.New()),
-		Clientset: acicnioperatorclient,
-		Informer:  informer,
-		Queue:     queu,
-		Handlers:  &OperatorHandler{},
+		Operator_Clientset: acicnioperatorclient,
+		K8s_Clientset: k8sclient,
+		Informer_Operator:  aci_operator_informer,
+		Informer_Deployment: aci_deployment_informer,
+		Informer_Daemonset: aci_daemonset_informer,
+		Operator_Queue:     operator_queue,
+		Deployment_Queue: deployment_queue,
+		Daemonset_Queue: daemonset_queue,
+		Resources: AciResources{},
 	}
 
 	log.Info("Adding Event Handlers")
-	informer.AddEventHandler(cache.ResourceEventHandlerFuncs{
+	aci_operator_informer.AddEventHandler(cache.ResourceEventHandlerFuncs{
 		AddFunc: func(obj interface{}) {
 			key, err := cache.MetaNamespaceKeyFunc(obj)
-			log.Infof("The acioperator key: %s", key)
+			log.Info("Added acicontainersoperator  key:", key)
 			if err == nil {
-				queu.Add(key)
+				operator_queue.Add(key)
 			}
 		},
 		UpdateFunc: func(prevObj, currentObj interface{}) {
 			key, err := cache.MetaNamespaceKeyFunc(currentObj)
-			log.Infof("Updated acicontainersoperator: %s", key)
+			log.Debug("Updated acicontainersoperator key:", key)
 			if err == nil {
-				queu.Add(key)
+				operator_queue.Add(key)
 			}
 		},
 		DeleteFunc: func(obj interface{}) {
 			key, err := cache.DeletionHandlingMetaNamespaceKeyFunc(obj)
-			log.Infof("Deleted acicontainersoperator key is : %s", key)
+			log.Info("Deleted acicontainersoperator key:", key)
+			if err == nil {
+				operator_queue.Add(key)
+			}
+		},
+	})
+
+	aci_deployment_informer.AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc: func(obj interface{}) {
+			key, err := cache.MetaNamespaceKeyFunc(obj)
+			log.Debug("Added Deployment key: %s", key)
+			if err == nil {
+				deployment_queue.Add(key)
+			}
+		},
+		UpdateFunc: func(prevObj, currentObj interface{}) {
+			key, err := cache.MetaNamespaceKeyFunc(currentObj)
+			log.Debug("Updated Deployment: %s", key)
+			if err == nil {
```

Comment:
do you want to check here if there is any change in prevObj and currentObj

Reply:
Noted.. providing check